tardis/io/util.py

```diff
@@ -240,6 +238,9 @@ def to_hdf(self, file_path, path='', name=None):
         data = self.get_properties()
         buff_path = os.path.join(path, name)
         self.to_hdf_util(file_path, buff_path, data)
+        for property in self.class_properties.keys():
```

Comment:
I'm thinking, would it be possible to call the `to_hdf` method during `to_hdf_util`? All complex objects we want to save should implement the `to_hdf` method. So during the writing process, you should be able to save the object (use ducktyping there).